### PR TITLE
[FIX] check if OpenGL was properly initialized before calling initial…

### PR DESCRIPTION
**Pull request by timosachsenberg**
…izeGL manually

### CODE REVIEW

**Comment by timosachsenberg** [comment]
the newer mac driver seem to be more sensitive in regards to calling opengl functions before proper initialization. This should fix the issue.

**Comment by timosachsenberg** [comment]
fixes https://github.com/OpenMS/OpenMS/issues/3413 on my mac

**Comment by timosachsenberg** [comment]
tested on linux - also works

**Comment by timosachsenberg** [comment]
might also fix https://github.com/OpenMS/OpenMS/issues/3354

File: src/openms_gui/source/VISUAL/Spectrum3DCanvas.cpp [hunk 1]
```diff
@@ -192,6 +192,10 @@ namespace OpenMS
   {
 #endif
 
+    // make sure OpenGL already properly initialized
+    QOpenGLContext *ctx = QOpenGLContext::currentContext();
+    if (!ctx || !ctx->isValid()) return;
```
Comment (hroest) [inline src/openms_gui/source/VISUAL/Spectrum3DCanvas.cpp]: how do we deal with failure here? according to the docs:

```
The typical way to handle context loss scenarios in applications is to check via this function whenever makeCurrent() fails and returns false. If this function then returns false, recreate the underlying native OpenGL context by calling create(), call makeCurrent() again and then reinitialize all OpenGL resources.
```
http://doc.qt.io/qt-5/qopenglcontext.html#isValid

Comment (timosachsenberg) [reply src/openms_gui/source/VISUAL/Spectrum3DCanvas.cpp]: I don’t think we need to at this point. I think we now just ignore premature update calls.

**Comment by hroest** [comment]
works on Linux

**Pull request merged**